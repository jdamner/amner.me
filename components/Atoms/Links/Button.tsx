import React from "react";

import { font } from "utils/header-font";

export default function Button(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >,
) {
  const newProps = {
    ...props,
    className: `${props.className} ${ButtonClasses.join(" ")}`,
  };
  return <button {...newProps} />;
}

/**
 * Button Classes
 *
 * @type {string[]}
 */
export const ButtonClasses: string[] = [
  "inline-block",
  "border-2",
  "border-orange-600",
  "bg-orange-600",
  "text-black",
  "uppercase",
  "text-2xl",
  "font-black",
  "px-4",
  "py-2",
  "hover:bg-orange-100",
  "hover:text-orange-600",
  "hover:border-orange-600",
  font.className,
];
