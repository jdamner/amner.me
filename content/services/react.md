---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
