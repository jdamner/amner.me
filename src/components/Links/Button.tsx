import React from "react";
/**
 * Button 
 * 
 * @param {JSX.IntrinsicElements['button']} props
 * @returns {JSX.Element}
 */
export default function Button(props: React.JSX.IntrinsicElements['button']): React.JSX.Element {

	const classes = ButtonClasses;
	props.className && classes.push(props.className);
	return (
		<button {...props} className={ButtonClasses.join(' ')} >
			{props.children}
		</button>
	);
}

/**
 * Button Classes
 * 
 * @type {string[]}
 */
export const ButtonClasses: string[] = [
	'border-2 border-slate-500',
	'uppercase',
	'font-bold',
	'bg-white',
	'dark:bg-slate-900',
	'px-4',
	'py-2',
	'text-slate-800',
	'dark:text-slate-200',
	'hover:bg-slate-100',
	'dark:hover:bg-slate-800',
	'hover:text-black',
	'dark:hover:text-white'
];