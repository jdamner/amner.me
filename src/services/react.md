---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.