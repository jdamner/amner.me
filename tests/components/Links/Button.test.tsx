import { render } from "@testing-library/react"
import Button from "../../../src/components/Links/Button"

describe("Button", () => {
  it("renders correctly", () => {
    const { container } = render(<Button>Test</Button>)
    expect(container).toMatchSnapshot()
  })
})
