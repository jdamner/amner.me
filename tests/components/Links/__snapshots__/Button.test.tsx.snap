// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="border-solid border-2 border-slate-800 uppercase font-black font-sans bg-white px-4 py-2 text-slate-800 hover:bg-slate-100 hover:text-black"
  >
    Test
  </button>
</div>
`;
